dotnetv3/Cognito/Actions/CognitoWrapper.cs

```diff
@@ -0,0 +1,362 @@
+            AuthParameters = authParameters,
+            AuthFlow = AuthFlowType.USER_PASSWORD_AUTH,
+        };
+
+        var response = await _cognitoService.AdminInitiateAuthAsync(request);
+        return response.Session;
+    }
+    // snippet-end:[Cognito.dotnetv3.AdminInitiateAuth]
+
+    // snippet-start:[Cognito.dotnetv3.InitiateAuth]
+    /// <summary>
+    /// Initiate authorization.
+    /// </summary>
+    /// <param name="clientId">The client Id of the application.</param>
+    /// <param name="userName">The name of the user who is authenticating.</param>
+    /// <param name="password">The password for the user who is authenticating.</param>
+    /// <returns>The response from the call to InitiateAuthAsync</returns>
+    public async Task<InitiateAuthResponse> InitiateAuthAsync(string clientId, string userName, string password)
+    {
+        var authParameters = new Dictionary<string, string>();
+        authParameters.Add("USERNAME", userName);
+        authParameters.Add("PASSWORD", password);
+
+        var authRequest = new InitiateAuthRequest
+
+        {
+            ClientId = clientId,
+            AuthParameters = authParameters,
+            AuthFlow = AuthFlowType.USER_PASSWORD_AUTH,
+        };
+
+        var response = await _cognitoService.InitiateAuthAsync(authRequest);
+        Console.WriteLine($"Result Challenge is : {response.ChallengeName}");
+
+        return response;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.InitiateAuth]
+
+    // snippet-start:[Cognito.dotnetv3.ConfirmSignUp]
+    /// <summary>
+    /// Confirm that the user has signed up.
+    /// </summary>
+    /// <param name="clientId">The Id of this application.</param>
+    /// <param name="code">The confirmation code sent to the user.</param>
+    /// <param name="userName">The user name.</param>
+    /// <returns></returns>
+    public async Task<bool> ConfirmSignupAsync(string clientId, string code, string userName)
+    {
+        var signUpRequest = new ConfirmSignUpRequest
+        {
+            ClientId = clientId,
+            ConfirmationCode = code,
+            Username = userName,
+        };
+
+        var response = await _cognitoService.ConfirmSignUpAsync(signUpRequest);
+        if (response.HttpStatusCode == HttpStatusCode.OK)
+        {
+            Console.WriteLine($"{userName} was confirmed");
+            return true;
+        }
+        return false;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.ConfirmSignUp]
+
+    // snippet-start:[Cognito.dotnetv3.ConfirmDevice]
+    /// <summary>
+    /// Initiates and confirms tracking of the device.
+    /// </summary>
+    /// <param name="accessToken">The user's access token.</param>
+    /// <param name="deviceKey">The key of the device from Amazon Cognito.</param>
+    /// <param name="deviceName">The device name.</param>
+    /// <returns></returns>
+    public async Task<bool> ConfirmDeviceAsync(string accessToken, string deviceKey, string deviceName)
```

Comment:
This method is not called, should it be part of the scenario?

Reply:
It was a "bonus" action to add code to confirm a device. I included the method in Actions but did not include code to confirm a device.